web/client/components/shapefile/ShapefileUploadAndStyle.jsx

```diff
@@ -81,6 +109,20 @@ class ShapeFileUploadAndStyle extends React.Component {
 
     getGeomType = (layer) => {
         if (layer && layer.features && layer.features[0].geometry) {
+            let unique;
```

Comment:
I was thinking to something similar to this (not tested):

```
const type = layer.features.reduce((previous, f) => {
    const currentType = getGeometryType(f.geometry);
    return currentType === previous ? previous : 'GeometryCollection';
}, null);

getGeometryType = (geometry) => {
    if (geometry && geometry.type === 'GeometryCollection') {
        return geometry.geometries.reduce((previous, g) => {
            if (g && g.type === previous) {
                return previous;
            }
            return g.type;
        }, null);
    }
    if (geometry) {
        return geometry.type;
    }
    return null;
}
```

Reply:
it seems that it always returns 'GeometryCollection'. I'm going to check